bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/AbstractRetryingOperation.java

```diff
@@ -301,15 +301,19 @@ protected Long getNextBackoff() {
 
   @VisibleForTesting
   public boolean inRetryMode() {
-    return currentBackoff != null;
+    // ResetStatusBasedBackoff will create the first attempt which creates currentBackoff with 0
+    // attempts.
+    return currentBackoff != null && currentBackoff.getAttemptCount() > 0;
   }
 
   /**
    * Either a response was found, or a timeout event occurred. Reset the information relating to
    * Status oriented exception handling.
    */
   protected void resetStatusBasedBackoff() {
-    this.currentBackoff = null;
+    // Reset the backoff parameters. CreateFirstAttempt will log the current time as the time of
+    // first call and enforce timeout from this timeOfFirstCall.
+    this.currentBackoff = exponentialRetryAlgorithm.createFirstAttempt();
```

Comment:
After our conversation, I looked at the code and I think the original reason this was null was set to null was a performance optimization. This new change will create new TimedAttemptSettings  for every message.

I dont anticipate this adding too much overhead, but please note this in the PR description for future reference

Reply:
Also please note in the description that there is behavioral change here: the first rpc counts as the first attempt